### PR TITLE
azure db discovery helm chart

### LINKED ISSUE #16019: Add Azure discovery to helm chart
Add helm chart for Azure config in the `db_service` section, like we have for AWS.

Please also update the docs here: docs/pages/setup/helm-reference/teleport-kube-agent.mdx

Related PRs: #15629 #15986

Issue this supports: #14688 

example config:
```yaml
  # Matchers for registering Azure-hosted databases.
  azure:
    # Database types. Valid options are:
    # 'mysql' - discovers and registers Azure MySQL single-server databases.
    # 'postgres' - discovers and registers Azure PostgreSQL single-server databases.
  - types: ["mysql", "postgres"]
    # Azure regions to register databases from. Valid options are:
    # '*' - discovers databases in all regions (default).
    # Any valid Azure region name. List all valid regions using the Azure "az" cli: `az account list-locations -o table`
    regions: ["eastus", "westus"]
    # Azure subscription IDs to register databases from. Valid options are:
    # '*' - discovers databases in all subscriptions (default).
    subscriptions: ["11111111-2222-3333-4444-555555555555"]
    # Azure resource groups to register databases from. Valid options are:
    # '*' - discovers databases in all resource groups within configured subscription(s) (default).
    resource_groups: ["group1", "group2"]
    # Azure resource tags to match when registering databases.
    tags:
      "*": "*"
```

What problem does this solve?
* Configuring Teleport to use azure discovery via helm chart

### PR DESCRIPTION
**Pull request by GavinFrazar**
Closes #16019

This PR adds azure discovery config to the helm chart.
Also includes:
- lint
- test
- schema update
- docs update for chart reference

### CODE REVIEW

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (ibeckermayer) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:365]: Nit: for some reason I can't articulate this sounds weird to me. Take it or leave it:
```suggestion
    YAML is very sensitive to correct spacing. When specifying lists in a `values.yaml` file, you might find it helpful
    to [use a linter](https://codebeautify.org/yaml-validator) to validate your YAML list and ensure that it is correctly formatted.
```

Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:365]: Sounds good, I actually copied this from the above awsDatabases doc, so I'll change that too

Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx]: oh it's actually in several places, I'll make it an include snippet

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:355]: I think we should add `roles: db` here as it is required for the feature to work. Maybe `extraEnv` also.

I don't think you need to bother with the `--set` tab, using azure database requires too many arguments, this is error-prone and I doubt anyone will do it.

File: examples/chart/teleport-kube-agent/README.md
Comment (hugoShaka) [inline examples/chart/teleport-kube-agent/README.md:218]: If I follow the README I'll end up with a non working deployment because this does not mentions azure authentication (so does the AWS one). Maybe it would be better to get rid of the command in the README and link to the reference as it already explains how this parameter works and what is required for the pod to authenticate to azure.

File: docs/pages/includes/yaml-lint-note.mdx
Comment (hugoShaka) [inline docs/pages/includes/yaml-lint-note.mdx:1]: @ptgott should we get rid of this admonition? It looks more like something that should be in a guide than in a reference.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:340]:
```suggestion

The default for each of these optional settings is `*`, which will auto-discover in all
subscriptions, regions, or resource groups accessible by the Teleport service
principal in Azure.

```

**Review by hugoShaka — COMMENTED** [review]
Code looks good to me, I made a couple of suggestions regarding the docs.

I'm sorry for opening some discussions not directly related to the PR, I know it was like this before and you just copied what we did for AWS databases.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:326]: Will update. Also, I think `dbResources` is incorrect - shouldn't it be `databaseResources`?

Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:326]: Yup, I renamed it at some point and forgot some mentions, the correct one is `databaseResources`

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:371]: Yeah I noticed that we didn't include all required fields in our examples, e.g. proxyAddr.

I'll provide a full working example config with all required fields.

> I don't think you need to bother with the --set tab, using azure database requires too many arguments, this is error-prone and I doubt anyone will do it.

Agreed, I think it would be easier to provide a shell here-doc yaml example than using the very limited --set syntax, or just provide only the yaml file example.

I provided --set only because that was the convention for other examples, but if we're ok with breaking convention I'll remove it (I really doubt anyone wants to use --set as you mentioned).

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:338]: will update then

**Comment by GavinFrazar** [comment]
@ibeckermayer @hugoShaka anything else for this one?

File: examples/chart/teleport-kube-agent/README.md
Comment (hugoShaka) [inline examples/chart/teleport-kube-agent/README.md:187]: Nit: we removed the command from the README so the `$VAR`syntax is not useful anymore, we could provide the value name instead.

**Review by hugoShaka — APPROVED** [review]

**Review by ibeckermayer — APPROVED** [review]
I'm not really qualified to comment on the helm charts themselves, but nothing jumps out at me

File: docs/pages/includes/yaml-lint-note.mdx
Comment (alexfornuto) [reply docs/pages/includes/yaml-lint-note.mdx:1]: IMHO it could only help or not help, but it doesn't hurt anything being where it is in `teleport-kube-agent.mdx`.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:289]:
```suggestion
  This section configures Azure database auto-discovery. You can configure databases for other platforms using the [`awsDatabases`](#awsdatabases) or [`databases`](#databases) sections.
```

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:463]: As noted above:

```suggestion
| `list` | `[]` | When the `db` chart role is used at least one of `databases`, `awsDatabases`,<br/> `azureDatabases`, `databaseResources` is required. |
```

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:286]: Why remove the `--set` version of this and the other edited sections?

**Review by alexfornuto — CHANGES_REQUESTED** [review]
Some questions and suggestions for the documentation portion of this PR.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:286]: The `--set` tab was extremely cumbersome to use for config - it seems misleading to even suggest someone use that method for configuration. Hugo suggested removing it and I agreed that no one is realistically going to bother with --set here. I don't care if we bring it back though, so it's up to you

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx
Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:330]: `[array]` - will change

Comment (GavinFrazar) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:330]: Oh, actually we show the type to be `list` above and the default value is `[]`. We do the same throughout this doc. I think list is clear enough.

File: examples/chart/teleport-kube-agent/README.md
Comment (GavinFrazar) [reply examples/chart/teleport-kube-agent/README.md]: will do

File: docs/pages/includes/yaml-lint-note.mdx
Comment (ptgott) [reply docs/pages/includes/yaml-lint-note.mdx:1]: As far as I know, Helm returns YAML parsing errors to the user in `helm install`, so it acts as the linter that this Admonition is recommending that you use. I only think we need this if there's a values file setting that is particularly tricky to write as YAML.

**Comment by GavinFrazar** [comment]
@alexfornuto I think everything is resolved, if you want we can restore the `--set` stuff but Hugo suggested just removing it and I agree, it's not something anyone will actually try to use because there's too many arguments to provide and it's error prone.

**Comment by GavinFrazar** [comment]
@alexfornuto just waiting on your approval, was there anything else you wanted addressed?

**Review by alexfornuto — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**
